api/tests/routes/public/collective/endpoints/patch_collective_offer_test.py

```diff
@@ -1074,7 +1074,12 @@ def assert_offer_venue_has_been_updated(self, client, api_key, venue_provider, s
         # check that offerVenue has been updated with dst values:
         # all common values have been copied and other are null.
         assert dst.keys() <= offer.offerVenue.keys()
-        assert all(offer.offerVenue[key] == value for key, value in dst.items())
+        for key, value in dst.items():
+            if key == "otherAddress" and value is None:
+                # we accept null for otherAddress but we store "" as our GET schemas expect a string
+                assert offer.offerVenue[key] == ""
+            else:
+                assert offer.offerVenue[key] == value
```

Comment:
Je n'aime pas vraiment l'idée de mettre de la logique (for/if) dans des tests. Ne pourrait-on pas changer `setup_and_send_request` plutot ?

Reply:
et/ou faire 2 tests

Reply:
j'ai essayé de faire au mieux avec la structure présente, dans l'idée il y a même des cas de départ incorrects (on setup une offre avec otherAddress à None, ce qui est invalide)

pour fixer rapidement le bug j'ai préféré toucher au moins de chose possible

Reply:
j'ai mis à jour le test finalement, tu me diras si c'est mieux